src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -1354,14 +1355,39 @@ protected void calculateLengthFromGeometry () {
                 e.setFloatingVehicleDropoffSuitability(getFloatingVehicleDropoffSuitability());
             }
         } else {
-            if (((TemporarySplitterVertex) v).isEndVertex()) {
-                e1 = new TemporaryPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
-                e1.setNoThruTraffic(this.isNoThruTraffic());
-                e1.setStreetClass(this.getStreetClass());
+            if (createSemiPermanentEdges) {
+                e1 = new SemiPermanentPartialStreetEdge(this, (StreetVertex) fromv, v, geoms.first, name);
+                e2 = new SemiPermanentPartialStreetEdge(this, v, (StreetVertex) tov, geoms.second, name);
```

Comment:
Why are two edges created here, but only one in each of the below conditions? Not criticizing the method, I just don't understand the situations where this is used. Perhaps a comment is warranted.

Reply:
The TemporarySplitterVertex is primarily used for linking the origin and destination to a StreetEdge. There is no need to create an edge from the street network to the origin and likewise no need for an edge from the destination to the street network as both traversals will never result in an optimal graph traversal. Travel from the origin to the destination always travels along the respective directed edges. 

This specificity also occurs in the when creating the TemporaryFreeEdges that connect the TemporarySplitterVertex to the Origin and Destination. See here: https://github.com/ibi-group/OpenTripPlanner/blob/a29d663dabef6e91d1bc2c079af9c02443dcc50b/src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java#L611-L634

Reply:
I'll add some comments.